fuel-core/src/database.rs

```diff
@@ -323,10 +329,23 @@ impl InterpreterStorage for Database {
     }
 
     fn coinbase(&self) -> Result<Address, Error> {
-        let height = self.get_block_height()?.unwrap_or_default();
-        let id = self.block_hash(height.into())?;
-        let block = self.storage::<FuelBlocks>().get(&id)?.unwrap_or_default();
-        Ok(block.header.producer)
+        let block = self.get_current_block()?.unwrap_or_else(|| {
+            std::borrow::Cow::Owned(FuelBlockDb::fix_me_default_block())
+        });
+        match block.consensus_type() {
+            ConsensusType::PoA => {
+                // FIXME: Get producer address from block signature.
+                // block_id -> Signature
+                let signature = Signature::default();
+                let message = unsafe {
+                    fuel_core_interfaces::common::fuel_crypto::Message::from_bytes_unchecked(
+                block.header.hash().into(),
+            )
+                };
```

Comment:
Formatting + use cached block id instead of recalculating it every time

```suggestion
                let message = unsafe {
                    fuel_core_interfaces::common::fuel_crypto::Message::from_bytes_unchecked(
                    block.header.id().into(),
                    )
                };
```

Reply:
It just cam from the database though so it will never be set and using the cached one will clone it.